src/wfc/collapse.rs

```diff
@@ -198,11 +203,24 @@ pub fn collapse(
         rules,
         &output_graph.edges,
         init,
-        output_graph.vertices,
-        iterations // 🐯
+        output_graph.vertices.clone(),
+        iterations, // 🐯
+        false
     );
-    output_graph.vertices = collapsed_vertices;
-    output_graph
+
+    // return a list of output graphs
+    collapsed_vertices
+        .into_iter()
+        .fold(Vec::new(), |mut graphs, vertices| {
+            graphs.push(
+                Graph::new(
+                    vertices.clone(),
+                    output_graph.edges.clone(),
+                    output_graph.all_labels.clone()
+                )
```

Comment:
Constructing a graph each time here serves no purpose because later on we only use the vertices from the graph, that means every clone of either `edges` or `all_labels` here is not needed. I think returning just `collapsed_vertices` would be fine.